lib/evaluate/intrinsics.h

```diff
@@ -15,12 +15,56 @@
 #ifndef FORTRAN_EVALUATE_INTRINSICS_H_
 #define FORTRAN_EVALUATE_INTRINSICS_H_
 
+#include "call.h"
+#include "type.h"
 #include "../common/idioms.h"
+#include "../parser/message.h"
+#include <memory>
+#include <optional>
+#include <ostream>
+#include <vector>
 
 namespace Fortran::evaluate {
 
-// Placeholder
-ENUM_CLASS(IntrinsicProcedure, IAND, IEOR, IOR, LEN, MAX, MIN)
+// Intrinsics are distinguished by their names and the characteristics
+// of their arguments, at least for now.
+using IntrinsicProcedure = const char *;  // not an owning pointer
 
+class Argument;
+
+struct CallCharacteristics {
+  parser::CharBlock name;
+  const Arguments &argument;
```

Comment:
Is `arguments` a better name for this field?

Reply:
Ok.